src/marqo/s2_inference/s2_inference.py

```diff
@@ -50,17 +53,65 @@ def _create_model_cache_key(model_name: str, device: str) -> Tuple:
     Returns:
```

Comment:
Why are we encoding the key, rather than keeping it a string?

Also, can we make the key "modelname_device_<PROPERTIES HASH OR HASH SUBSTRING>" or something similar (readibility would help)